src/v2.rs

```diff
@@ -1587,7 +1672,10 @@ impl EszipV2 {
 
     // Import map must be either JSON or JSONC (but JSONC is a special case;
     // it's allowed when embedded in a Deno's config file)
-    if import_map.kind == ModuleKind::JavaScript {
+    if !matches!(
+      import_map.kind,
+      ModuleKind::Json | ModuleKind::Jsonc | ModuleKind::OpaqueData
+    ) {
```

Comment:
I kept the same logic here, though I feel like OpaqueData could be removed. The original condition was not good.